pkg/tests/schema_pulumi_test.go

```diff
@@ -1769,3 +1773,360 @@ resource "prov_test" "test" {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateDiagFunc: func(i interface{}, p cty.Path) diag.Diagnostics {
+								return diag.Errorf("Error")
+							},
+						},
+						"nested_other_prop": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateDiagFunc: func(i interface{}, p cty.Path) diag.Diagnostics {
+								return nil
+							},
+						},
+					},
+				},
+			},
+			expectedProps: map[string]interface{}{
+				"collectionProp": map[string]interface{}{
+					"nestedOtherProp": "ABC",
+					"nestedProp":      "ABC",
+				},
+			},
+			expectFailure: true,
+		},
+		{
+			name: "TypeSet Do not validate nested fields",
+			cloudVal: []interface{}{
+				map[string]interface{}{
+					"nested_prop": "ABC",
+				},
+			},
+			schema: schema.Schema{
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"nested_prop": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: func(i interface{}, s string) ([]string, []error) {
+								return []string{}, []error{errors.New("Error")}
+							},
+						},
+					},
+				},
+			},
+			expectedProps: map[string]interface{}{
+				"collectionProps": []interface{}{
+					map[string]interface{}{
+						"nestedProp": "ABC",
+					},
+				},
+			},
+			expectFailure: true,
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			resMap := map[string]*schema.Resource{
+				"prov_test": {
+					Schema: map[string]*schema.Schema{
+						"collection_prop": &tc.schema,
+						"other_prop": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+					Importer: &schema.ResourceImporter{
+						StateContext: func(ctx context.Context, rd *schema.ResourceData, i interface{}) ([]*schema.ResourceData, error) {
+							err := rd.Set("collection_prop", tc.cloudVal)
+							assert.NoError(t, err)
+							err = rd.Set("other_prop", "test")
+							assert.NoError(t, err)
+							return []*schema.ResourceData{rd}, nil
+						},
+					},
+					ReadContext: func(ctx context.Context, rd *schema.ResourceData, i interface{}) diag.Diagnostics {
+						err := rd.Set("collection_prop", tc.cloudVal)
+						assert.NoError(t, err)
+						err = rd.Set("other_prop", "test")
+						assert.NoError(t, err)
+						return nil
+					},
+				},
+			}
+			tfp := &schema.Provider{ResourcesMap: resMap}
+			bridgedProvider := pulcheck.BridgedProvider(t, "prov", tfp, pulcheck.DisablePlanResourceChange())
+			program := `
+name: test
+runtime: yaml
+`
+			pt := pulcheck.PulCheck(t, bridgedProvider, program)
+			outPath := filepath.Join(pt.CurrentStack().Workspace().WorkDir(), "out.yaml")
+
+			imp := pt.Import("prov:index/test:Test", "mainRes", "mainRes", "", "--out", outPath)
+			tc.expectedProps["otherProp"] = "test"
+
+			contents, err := os.ReadFile(outPath)
+			assert.NoError(t, err)
+			expected := PulumiYaml{
```

Comment:
This is cool, do we think this is better than inline YAML? Should we standardise on this and move some utility functions/types into a shared folder for pulumi yaml?

Reply:
I personally think it is better, especially for cases like this where you are dynamically updating the yaml. Would it make sense to do it in the `providertest` repo? I can create an issue.

Reply:
Yup, def handy for edits. I guess the downside is that the test is no longer "just" schema + yaml program.

Reply:
Yea proper Go code is friendlier to the IDE.